sycl/source/CMakeLists.txt

```diff
@@ -87,6 +87,7 @@ set(SYCL_SOURCES
 )
 
 add_sycl_rt_library(sycl ${SYCL_SOURCES})
+target_include_directories(sycl PRIVATE "detail/include")
```

Comment:
@AlexeySachkov and what about sycl**d**? I suggest putting this inside add_sycl_rt_library and using ${LIB_NAME} as target name. Otherwise you break windows builds.